fission-web-server/README.md

```diff
@@ -52,11 +52,23 @@ To mimic the full "fission stack" for local development, you can use the include
    -  `docker compose exec dns-auth pdnsutil create-zone runfission.test`
    -  `docker compose exec dns-auth pdnsutil create-zone fissionuser.test`
    -  `docker compose exec dns-auth pdnsutil create-zone fissionapp.test`
-4. Point your local DNS resolver to localhost.
+4. Create some initial DNS records:
+   - `docker compose exec dns-auth pdnsutil add-record runfission.test. @ A "127.0.0.1"`
+   - `docker compose exec dns-auth pdnsutil add-record fissionuser.test. gateway A "127.0.0.1"`
+   - `docker compose exec dns-auth pdnsutil add-record fissionapp.test. gateway A "127.0.0.1"`
+5. Point your local DNS resolver to localhost.
    - on macOS: this is under System Preferences > Network > Advanced.
    - on Linux: Add `nameserver 127.0.0.1` to `/etc/resolv.conf`
-  
-You can now build / run the haskell server. The included `server.yaml.example` is configured to work with the local docker setup.
+6. Pin the CID for the new app placeholder:
+   `docker compose exec ipfs ipfs pin add -r QmRVvvMeMEPi1zerpXYH9df3ATdzuB63R1wf3Mz5NS5HQN`
+7. Copy `addon-manifest.json.example` to `addon-manifest.json` and change `CHANGE_ME_TO_NUMBER` to a random number (e.g. 42)
+8. Copy `server.yaml.example` to `server.yaml`
+9. Build / install the server (if using nix-shell you can run `server-install`)
+10. Run the server (if using nix-shell you can run `server-debug`)
+   
+You should now have a rest server running on port `1337` (test: http://runfission.test:1337/ping).
+
+You can use the fission CLI to register users and deploy apps: `fission setup -R runfission.test:1337` (always pass the `-R runfission.test:1337` to commands).
```

Comment:
Okay, I believe the issue with ipfs not starting is due to the network being missing from the docker compose.

```
  ipfs:
    image: ipfs/go-ipfs:v0.11.0
    ports:
      - "4001:4001"
      - "5001:5001"
      - "8080:8080"
    volumes:
      - ./data/ipfs:/data/ipfs
    networks:
      - default
```

Though I'm having the following issue when I try to stop and start the containers with this change included.

```
»  fission -R runfission.test:1337 setup
🚫 Problem fetching Fission's DID
```

Reply:
Looks like this...

https://github.com/ipfs/go-ipfs/issues/8645

Reply:
Yep, modified ./data/ipfs/config with the workaround at https://github.com/ipfs/go-ipfs/issues/8645#issuecomment-1113905617 and the daemon starts up now.

Reply:
I get the following when trying to set up a new account after I have everything working.

```
»  fission -R runfission.test:1337 setup
🌱 Setting up environment
🪐 Downloading managed IPFS for macOS
🕖🎛️  Configuring managed IPFS
🔑 Setting up keys
🏠 Do you have an existing account? [Y/n] n
Username: justin
Email: justin@justinjohnson.io
🚫 It looks like that account already exists. Please try again or contact Fission support at https://fission.codes
```

Logs have the following

```
2022-06-20 11:09:50.723688: [debug] (SQL) SELECT "id","public_key","exchange_keys","email","username","role","active","verified","data_root","data_root_size","heroku_add_on_id","secret_digest","inserted_at","modified_at" FROM "user" WHERE "username"=?; [PersistText "justin"]
@(<unknown>:0:0)
2022-06-20 11:09:50.726376: [debug] (SQL) SELECT "id","public_key","exchange_keys","email","username","role","active","verified","data_root","data_root_size","heroku_add_on_id","secret_digest","inserted_at","modified_at" FROM "user" WHERE "username"=?; [PersistText "justin"]
@(<unknown>:0:0)
2022-06-20 11:09:50.728027: [debug] (SQL) SELECT "id","public_key","exchange_keys","email","username","role","active","verified","data_root","data_root_size","heroku_add_on_id","secret_digest","inserted_at","modified_at" FROM "user" WHERE "public_key"=?; [PersistText "URQ3zxrfFFJRyhcWAdpcqSST3UuYVecW+DgIsq6UR/4="]
@(<unknown>:0:0)
2022-06-20 11:09:50.732241: [error] Username justin already exists
@(library/Fission/Web/Server/Error.hs:59:3)
```

But I don't see the user in the db

```
»  docker compose exec postgres psql -U postgres -d web_api
psql (11.16)
Type "help" for help.

web_api=# select * from user;
   user   
----------
 postgres
(1 row)
```

I have no postgres running locally, and only one in docker, so I think I'm in the right db.

Reply:
> I have no postgres running locally, and only one in docker, so I think I'm in the right db.

try `SELECT * FROM "user"` (quotes important) ... it looks like it's pulling from the PG user db ... unfortunate name conflict :-/

Reply:
Per out Zoom call, I'm assigning this back to you.  (I believe it was assigned to me just for review purposes.)  Additional items we discussed...

- peer id being wrong in server.yaml, need to grab from the ipfs node running in docker
- send_in_blue api key for email verification, will actually send email as no stubbed out
- ipfs not starting due to MDNS issue on M1, workaround at https://github.com/ipfs/go-ipfs/issues/8645#issuecomment-1113905617, but sounded like fixed in latest ipfs in docker, and I wasn't running your branch because I thought it was just README updates

Reply:
:shipit: